pytorch_to_returnn/torch/nn/modules/linear.py

```diff
@@ -123,6 +122,41 @@ def create_returnn_layer_dict(self, *inputs: Tensor, **kwargs) -> Dict[str, Any]
     return {"class": "dot", "red1": red1, "red2": red2, "var1": var1, "var2": var2, "from": sources}
 
 
+class DotProductAlongDim(Module):
+  """
+  Maps to RETURNN DotLayer for a special case
+  """
+  is_original_torch_module = False
+
+  def __init__(self, dim: int):
+    super(DotProductAlongDim, self).__init__()
+    self.dim = dim
+
+  def _get_output_shape_from_returnn(self, inputs_flat: List[Tensor], layer: LayerBase
+                                     ) -> Tuple[Tuple[int, ...], Dict[int, int]]:
+    assert len(inputs_flat) == 2
+    input1, input2 = inputs_flat
+    assert all(isinstance(input_, Tensor) for input_ in [input1, input2])
+    assert all(len(input_.shape) >= 2 for input_ in [input1, input2]), "not implemented otherwise"
+
+    shape = input1.shape
+    assert input2.shape == shape, "not implemented otherwise"  # could e.g. consider broadcasting axes
+    del shape[self.dim]
```

Comment:
Did you test this code? This here would not work I think because `shape` is a tuple (thus not mutable). You need to convert it to `list` first.